src/client/src/containers/AzureSubscriptions/index.tsx

```diff
@@ -110,29 +118,69 @@ class AzureSubscriptions extends React.Component<Props, IState> {
     }
     return () => {};
   }
+
+  public getServicesOrganizer(
+    type: string,
+    isLoggedIn: boolean,
+    setDetailPage: any,
+    title: any
+  ) {
+    const { formatMessage } = this.props.intl;
+    return (
+      <div
+        className={classnames(styles.servicesContainer, {
+          [styles.overlay]: !isLoggedIn
+        })}
+      >
+        {azureServiceOptions.map(option => {
```

Comment:
I think this is setup a bit wrong.

Notice this is what happens if I change the type of both services to the same thing:
![image](https://user-images.githubusercontent.com/32103099/61014290-ba519580-a354-11e9-998a-103ae7c63595.png)

This would mean if you add an additional service of a type already provided you will encounter the same bug (For example: App service as a cloud hosting platform).

Instead I think your code should be something like the following
```ts
<div className={styles.servicesCategory}>
  <div className={styles.categoryTitle}>{type}</div>
  <div className={styles.categoryDescriptor}>{formatMessage(title)}</div>
  {azureServiceOptions.map(option => {
    if (option.type === type) {
      return (
        <div
          key={JSON.stringify(option.title)}
          className={classnames(styles.subscriptionCardContainer)}
        >
          <Card
            option={option}
            buttonText={this.addOrEditResourceText(option.internalName)}
            handleButtonClick={this.getServicesModalOpener(option.internalName)}
            disabled={!isLoggedIn}
            handleDetailsClick={setDetailPage}
            useNormalButtons={this.isSelectionCreated(option.internalName)}
          />
        </div>
      );
    }
  })}
</div>
```

Which is slightly better but the css is a bit messed up, as the cards should be beside each other but they're on a new line here. They should behave like the pages kind of. Where if the screen is too small it will move them to the next row, otherwise they will try to populate the same row. 

![image](https://user-images.githubusercontent.com/32103099/61014598-42846a80-a356-11e9-9988-27768235ed37.png)


Now you may notice that if you have an empty type, (a type defined with no elements the title will show up. If you want this to happen leave the above code as is. If not, you can do a filter, then a length check before rendering.

Reply:
Thanks for bringing this to my attention! I definitely should have tested/accounted for those cases. I made the changes you suggested and to account for only showing the categories in use, I stored the unique option types so I can only render unique categories that have child elements

Reply:
![image](https://user-images.githubusercontent.com/32103099/61021035-cbf36700-a36d-11e9-852d-24e259be65be.png)
Looks better!